opencti-platform/opencti-front/src/private/components/analyses/reports/ReportPopoverDeletion.tsx

```diff
@@ -0,0 +1,130 @@
+import { graphql } from 'react-relay';
+import DialogContent from '@mui/material/DialogContent';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import React, { FunctionComponent, useState } from 'react';
+import { useTheme } from '@mui/styles';
+import { useNavigate } from 'react-router-dom';
+import DialogContentText from '@mui/material/DialogContentText';
+import { ReportPopoverDeletionQuery$data } from '@components/analyses/reports/__generated__/ReportPopoverDeletionQuery.graphql';
+import { QueryRenderer } from '../../../../relay/environment';
+import { useFormatter } from '../../../../components/i18n';
+
+import type { Theme } from '../../../../components/Theme';
+import Transition from '../../../../components/Transition';
+import useApiMutation from '../../../../utils/hooks/useApiMutation';
+
+const reportPopoverDeletionQuery = graphql`
+  query ReportPopoverDeletionQuery($id: String) {
+    report(id: $id) {
+      deleteWithElementsCount
+    }
+  }
+`;
+
+const reportPopoverDeletionMutation = graphql`
+  mutation ReportPopoverDeletionMutation($id: ID!, $purgeElements: Boolean) {
+    reportEdit(id: $id) {
+      delete(purgeElements: $purgeElements)
+    }
+  }
+`;
+
+interface ReportPopoverDeletionProps {
+  reportId: string;
+  displayDelete: boolean;
+  handleClose: () => void;
+  handleCloseDelete: () => void;
+}
+
+const ReportPopoverDeletion: FunctionComponent<ReportPopoverDeletionProps> = ({
+  reportId,
+  displayDelete,
+  handleClose,
+  handleCloseDelete,
+}) => {
+  const { t_i18n } = useFormatter();
+  const theme = useTheme<Theme>();
+  const navigate = useNavigate();
+  const [purgeElements, setPurgeElements] = useState(false);
+  const [deleting, setDeleting] = useState(false);
+  const [commitMutation] = useApiMutation(reportPopoverDeletionMutation);
+  const submitDelete = () => {
+    setDeleting(true);
+    commitMutation({
+      variables: { id: reportId, purgeElements },
+      onCompleted: () => {
+        setDeleting(false);
+        handleClose();
+        navigate('/dashboard/analyses/reports');
+      },
+    });
+  };
+  return (
+    <Dialog
+      open={displayDelete}
+      TransitionComponent={Transition}
+      PaperProps={{ elevation: 1 }}
+      onClose={handleCloseDelete}
+    >
+      <DialogContent>
+        <DialogContentText>
+          {t_i18n('Do you want to delete this report?')}
+        </DialogContentText>
+        <QueryRenderer
```

Comment:
QueryRenderer should not be used in new components as it is an old way of calling queries for class components. use hooks instead